src/test/java/org/folio/innreach/repository/ContributionCriteriaConfigurationRepositoryTest.java

```diff
@@ -0,0 +1,176 @@
+package org.folio.innreach.repository;
+
+import org.folio.innreach.domain.entity.CentralServer;
+import org.folio.innreach.domain.entity.ContributionCriteriaConfiguration;
+import org.folio.innreach.domain.entity.ContributionCriteriaStatisticalCodeBehavior;
+import org.folio.innreach.fixture.ContributionCriteriaConfigurationFixture;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.jdbc.Sql;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@Sql(scripts = "classpath:db/pre-populate-central-server.sql")
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+class ContributionCriteriaConfigurationRepositoryTest extends BaseRepositoryTest {
+
+  private static final String PRE_POPULATED_CENTRAL_SERVER_ID = "edab6baf-c696-42b1-89bb-1bbb8759b0d2";
+  private static final String PRE_POPULATED_CENTRAL_SERVER_CODE = "abc12";
+  private static final List<UUID> UUUID_IDs = List.of(UUID.randomUUID(),UUID.randomUUID(),UUID.randomUUID());
+
+  @Autowired
+  private CentralServerRepository centralServerRepository;
+
+  @Autowired
+  private ContributionCriteriaConfigurationRepository configurationRepository;
+
+  @Autowired
+  private ContributionCriteriaConfigurationRepository contributionCriteriaConfigurationRepository;
+
+  @BeforeEach
+  void beforeEach() {
+    CentralServer centralServer = centralServerRepository.fetchOne(UUID.fromString(PRE_POPULATED_CENTRAL_SERVER_ID)).get();
+    assertNotNull(centralServer);
+    ContributionCriteriaConfiguration criteriaConfiguration
+      = ContributionCriteriaConfigurationFixture
+      .createTestContributionCriteriaConfiguration(centralServer.getId());
+    contributionCriteriaConfigurationRepository.saveAndFlush(criteriaConfiguration);
+  }
+
+  @Test
+  void chekIfContributionCriteriaConfigurationCreated() {
+    var centralServer = centralServerRepository.fetchOne(UUID.fromString(PRE_POPULATED_CENTRAL_SERVER_ID)).get();
```

Comment:
Are you sure you need to fetch the CentralServer entity here?

Reply:
Done